### PR TITLE
Add RP2350 support

### PR DESCRIPTION
**Pull request by konkers**

### CODE REVIEW

**Comment by konkers** [comment]
Tested on both Pico 1 (rp2040) and Pico 2 (rp2350) using https://github.com/pyocd/pyOCD/pull/1718

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# RaspberryPi Pico RP2040 flash algorithm
+# RaspberryPi Pico RP2 flash algorithm
 
-This is a flash algorithm for the RP2040 chip, used in the Raspberry Pi Pico board. 
+This is a flash algorithm for the RP2040 and RP2350 chipx, used in the Raspberry Pi Pico board.
```
Comment (thejpster) [inline README.md]: typo

File: src/main.rs [hunk 2]
```diff
@@ -3,23 +3,112 @@
...
+}
+
+fn find_func<T>(tag: [u8; 2]) -> Result<T, ErrorCode> {
+    let tag = u16::from_le_bytes(tag) as u32;
+
+    /// This location in ROM holds a 3 byte magic value that confirms the validity of the
+    /// ROM as well as identifying different interfaces (referred here in as V1 and V2).
+    const BOOTROM_MAGIC: *const [u8; 3] = 0x0000_0010 as _;
+
+    /// Magic value for V1 ROMs.
```
Comment (thejpster) [inline src/main.rs]: Do you mean RP2040, or RP2350 A1?

Comment (konkers) [reply src/main.rs]: I mean RP2040 here but struggled with the right nomenclature.  The RP2040 datasheet defines the magic as `'M', 'u', 0x01` and  2350 datasheet defines the magic as `'M', 'u', 0x02`.  This is different from the byte at `0x13` which is the "bootrom" version.   I'm happy to change these to RP2040 and RP235x if that's clearer.

Comment (thejpster) [reply src/main.rs]: I'd call them RP2040 and RP2350 I think. Especially as each ROM has multiple revisions.

**Comment by thejpster** [comment]
Does it work on RISC-V mode? There are some subtle changes to the lookup code compared to Arm.

**Comment by konkers** [comment]
> Does it work on RISC-V mode? There are some subtle changes to the lookup code compared to Arm.

Completely untested in RISC-V mode.

File: README.md (hunk 1 above)
Comment (konkers) [reply README.md]: Thanks, my spelling is terrible :)

File: src/main.rs (hunk 2 above)
Comment (konkers) [reply src/main.rs]: renamed

**Review by thejpster — APPROVED** [review]

**Pull request merged**